src/jahongir/service/CategoryService.java

```diff
@@ -15,43 +15,58 @@ public Category[] getCategories() {
     @Override
     public boolean add(Object object) {
         Category category = (Category) object;
-
-        if (validate(category)) {}
-
-        return false;
+        if (validate(category)) {
+            categories[indexCategories++] = category;
+            return true;
+        } else {
+            return false;
+        }
     }
 
     @Override
     public boolean delete(UUID id) {
+
         return false;
     }
 
     @Override
     public boolean update(UUID id, Object object) {
         Category updateCategory = (Category) object;
-
         for (int i = 0; i < indexCategories; i++) {
             if (categories[i].getId().equals(id)) {
                 categories[i].setName(updateCategory.getName());
                 return true;
             }
         }
-
         return false;
     }
 
     @Override
     public Object[] list(UUID id) {
-        Category[] parentCategories = new Category[100];
-        int index = 0;
-
-        return parentCategories;
+        Category[] parentCategories = new Category[indexCategories];
+        int count = 0;
+        for (int i = 0; i < indexCategories; i++) {
+            if (categories[i].getId() != null && categories[i].getId().equals(id)) {
+                parentCategories[count++] = categories[i];
+            }
+        }
+        Category[] result = new Category[count];
```

Comment:
there is you can just return `parentCategory` why you need to create array 2 times?

botta siz bitta `parentCategory` ni oziniyam qaytarb qoysez boladi. nma kere 2 marta array yaratib?

Reply:
ok i understand the mistake and i will share you soon